src/server/scripts/Shadowlands/zone_exiles_reach.cpp

```diff
@@ -0,0 +1,548 @@
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Object.h"
+#include "Player.h"
+#include "ScriptMgr.h"
+#include "Transport.h"
+#include "Conversation.h"
+#include "TemporarySummon.h"
+#include "CreatureAIImpl.h"
+
+// Scripting in this section is from login to arriving on beach for alliance and horde
+
+enum HordeSparingPartner
+{
+    CONVERSATION_PREFIGHT       = 14422,
+    CONVERSATION_AGGRO          = 14423,
+    CONVERSATION_JUMP           = 14424,
+    EQUIPMENT_AXE               = 175161,
+    EVENT_MOVE_TO_A_POSITION    = 1,
+    EVENT_PREFIGHT_CONVERSATION = 2,
+    EVENT_WALK_BACK             = 3,
+    NPC_SPARING_PARTNER         = 166814,
+    NPC_SPAR_POINT_ADVERTISMENT = 174971,
+    NPC_GRUNT_THROG             = 166583,
+    QUEST_KILL_CREDIT           = 155607,
+    SPELL_JUMP_BEHIND           = 312757,
+    SPELL_COMBAT_TRAINING       = 323071,
+    SPELL_UPDATE_PHASE_SHIFT    = 82238,
+    SPELL_SUMMON_THROG          = 325107,
+    SAY_I_CONCEDE               = 0
+};
+
+struct npc_horde_sparring_partner : public ScriptedAI
+{
+    npc_horde_sparring_partner(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        SetEquipmentSlots(false, EQUIPMENT_AXE, EQUIP_NO_CHANGE, EQUIP_NO_CHANGE);
+    }
+
+    void IsSummonedBy(WorldObject* summonerWO) override
+    {
+        _jumped = false;
+
+        Unit* summoner = summonerWO->ToUnit();
+        if (!summoner)
+            return;
+
+        if (Creature* throg = me->FindNearestCreature(NPC_GRUNT_THROG, 5.0f))
+            _throgLocation = throg->GetPosition();
+
+        if (summoner->IsPlayer())
+        {
+            _playerGUID = summoner->GetGUID();
+            _events.ScheduleEvent(EVENT_MOVE_TO_A_POSITION, 2s);
+        }
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        if (!me->IsAlive())
+            return;
+
+        me->CombatStop(true);
+        EngagementOver();
+        me->ResetPlayerDamageReq();
+        _events.ScheduleEvent(EVENT_WALK_BACK, 1s);
+    }
+
+    void MovementInform(uint32 uiType, uint32 uiId) override
+    {
+        if (uiType != POINT_MOTION_TYPE)
+            return;
+
+        if (uiId == 1)
+        {
+            me->SetWalk(true);
+            me->GetMotionMaster()->MovePoint(2, me->GetFirstCollisionPosition(2.0f, (float)rand_norm() * static_cast<float>(2 * M_PI)));
+        }
+
+        if (uiId == 2)
+        {
+            if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                me->SetFacingToObject(player);
+            me->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_PC | UNIT_FLAG_UNINTERACTIBLE);
+            me->SetFaction(32); // *** BIG HACK to be removed after issue with entrying combat with faction 35 fixed ***
```

Comment:
This is not a bug, creatures can attack other creatures with faction 35

Reply:
Why not here?

Reply:
If I leave him at 35 player cannot attack.

Reply:
It's not a bug, but a missing implementation: any unit should be able to fight with friendly creatures, but there's no support for that. Also, that creature uses SummoProperties::Flags "Attackable by Summoner" which is also not implemented. Either a hack or no fix until proper implementation is added.

Reply:
I'm gonna work on the alliance side now. This is just a wip so hacks are allowed to get to the end. It's up to the devs if they want to fix the issues after that.